rest/tests/test_vocabulary.py

```diff
@@ -735,52 +793,112 @@ def get_facet_counts():
-            sorted(get_facet_counts().keys())
+            sorted(["course", "run", "resource_type", vocab_key]),
+            sorted(self.get_results()['facet_counts'].keys())
         )
 
         term1 = Term.objects.create(vocabulary=vocab, label="term1", weight=1)
         term2 = Term.objects.create(vocabulary=vocab, label="term2", weight=2)
 
         # Assign all types to vocab.
-        for resource in LearningResource.objects.all():
-            vocab.learning_resource_types.add(resource.learning_resource_type)
+        type_names = [t.name for t in LearningResourceType.objects.all()]
+        self.patch_vocabulary(self.repo.slug, vocab.slug, {
+            "learning_resource_types": type_names
+        }, skip_assert=True)  # Skip assert due to different order of types
 
         # No terms yet.
         self.assertEqual(
             sorted([t['key'] for t in
-                    get_facet_counts()[vocab.slug]['values']]),
+                    self.get_results()['facet_counts'][vocab_key]['values']]),
             []
         )
 
         resource1 = LearningResource.objects.all()[0]
-        resource1.terms.add(term1)
         resource2 = LearningResource.objects.all()[1]
-        resource2.terms.add(term2)
+        self.patch_learning_resource(self.repo.slug, resource1.id, {
+            "terms": [term1.slug]
+        })
+        self.patch_learning_resource(self.repo.slug, resource2.id, {
+            "terms": [term2.slug]
+        })
 
         # Vocab shows up because there are slugs here.
         self.assertEqual(
             sorted([t['key'] for t in
-                    get_facet_counts()[vocab.slug]['values']]),
+                    self.get_results()['facet_counts'][vocab_key]['values']]),
             sorted([term1.slug, term2.slug])
         )
+        self.assert_results([resource1], vocab_key, term1.slug)
+        self.assert_results([resource2], vocab_key, term2.slug)
 
         # Term is removed from facet list.
         self.delete_term(self.repo.slug, vocab.slug, term1.slug)
         self.assertEqual(
             sorted([t['key'] for t in
-                    get_facet_counts()[vocab.slug]['values']]),
+                    self.get_results()['facet_counts'][vocab_key]['values']]),
             sorted([term2.slug])
         )
+        self.assert_results([], vocab_key, term1.slug)
+        self.assert_results([resource2], vocab_key, term2.slug)
 
         self.delete_vocabulary(self.repo.slug, vocab.slug)
         # No vocabs left.
         self.assertEqual(
             sorted(["course", "run", "resource_type"]),
-            sorted(get_facet_counts().keys())
+            sorted(self.get_results()['facet_counts'].keys())
         )
+        self.assert_results([], vocab_key, term1.slug)
+        self.assert_results([], vocab_key, term2.slug)
+
+    def test_vocab_num_queries(self):
+        """Make sure number of queries is reasonable for vocab and term."""
```

Comment:
I know this test is really only to check that the query count doesn't go up, but I have to comment that 25, 20, and 37 queries are far from reasonable. I'm going to open a ticket to look into that.